Source/DafnyLanguageServer/Workspace/Notifications/VerificationDiagnosticsParams.cs

```diff
@@ -370,14 +370,23 @@ VersionedTextDocumentIdentifier DocumentIdentifier
   ) : VerificationTree("Document", DocumentIdentifier.Uri.ToString(), DocumentIdentifier.Uri.ToString(), DocumentIdentifier.Uri.ToString(),
     DocumentIdentifier.Uri.ToUri(),
     ComputeRange(Program, DocumentIdentifier.Uri.ToUri()), new Position(0, 0)) {
+    internal static Range ComputeRange(INode program, Uri uri) {
+      IToken? end = null;
+      if (program is Program dafnyProgram) {
+        end = dafnyProgram.DefaultModuleDef.RangeToken.StartToken;
+        while (end != null && end.Uri != uri) {
+          end = end.Next;
+        }
+      }
+      if (end == null) {
+        var fileNode = FindFileNode(program, uri);
```

Comment:
What was unreliable about the previous code? Using `dafnyProgram.DefaultModuleDef.RangeToken` seems unreliable to me. With projects there's no single main file. I'm not sure what the value of `dafnyProgram.DefaultModuleDef.RangeToken` will be.

Reply:
FindFileNode returns null after the first time. I couldn't understand what was wrong there. I found at least one problematic case, which is when the parser fails so there is strictly nothing in dafnyProgram, so looking at the first node with an adequate token is not robust.
So, I based my change on what the formatter is doing, which is to obtain the first token of a file based on DefaultModuleDecl.
If you have better ideas I'll happily consider them.

Reply:
I see.

I think the better solution would be for Program to contain a list of FileModuleDefinition, where each has a StartToken with file specific URI even if parsing fails. These could replace DefaultModuleDef at some point but initially could be just for the token purpose.

I don't think `dafnyProgram.DefaultModuleDef.RangeToken` will work when in project mode, since there's no root file.